### PR TITLE
refactor: a huge update

### PR DESCRIPTION
**Pull request by ModyQyW**
### Description 描述

这个 PR 相当大，做了以下的工作：

1. 更新了依赖，包括包依赖和 playground 依赖
2. 调整了项目结构，现在是一个 monorepo
3. 增加了一个包 @uni-helper/manifest-json-schema，该包将用于 uni-app-schema-vscode
4. 修复了类型定义，现在 manifest 相关类型定义都已经被更新

### Linked Issues 关联的 Issues

https://github.com/uni-helper/uni-app-schemas-vscode/issues/20

### Additional context 额外上下文

我不认为存在对外的破坏性更新，但存在对内的破坏性更新，该 PR 仍然需要仔细审查。

## Summary by CodeRabbit

- **New Features**
  - Introduced a Vite plugin for file system-based routing in uni-app applications.
  - Added new configuration interfaces for various mini programs and H5 applications.
  - Updated app configuration settings in `manifest.json` for the "uni-helper" Uni-app.
  - Added a JSON schema for `pages.json` in uni-app.

- **Enhancements**
  - Updated package manager to `pnpm@9.0.6`.
  - Enhanced ESLint and Prettier settings for better code styling and validation.
  - Upgraded dependencies and devDependencies for improved performance and security.

- **Documentation**
  - Added READMEs detailing new plugins and JSON schemas for uni-app.

- **Bug Fixes**
  - Removed `manifest.json` from `.gitignore` to ensure it is tracked and updated.

- **Chores**
  - Modified `pnpm-workspace.yaml` to include all packages in the `packages` directory.
  - Updated Node.js version to 20 in `.nvmrc`.

### CODE REVIEW

**Comment by ModyQyW** [comment]
@coderabbitai review

**Comment by zguolee** [comment]
牛牛牛，明天看一看

**Comment by ModyQyW** [comment]
如果没有进一步的评论，我可能会在两天内合并 🙏

File: packages/core/src/config/types/mpWeixin.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
+import type { PlatformUniStatistics } from './uniStatistics'
+
+export interface MpWeixin {
+  /** 微信小程序的 appid */
+  appid?: string
+
+  /**
+   * 微信小程序项目设置，详见 <https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html#setting>
+   */
+  setting?: {
```
Comment (zguolee) [inline packages/core/src/config/types/mpWeixin.ts:10]: mp-weixin.setting缺少urlCheck定义，https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html#setting

File: packages/core/src/config/types/appPlus.ts [hunk 2]
```diff
@@ -0,0 +1,2149 @@
...
+
+      /**
+       * 应用的CFBundleName名称
+       *
+       * @default "HBuilder"
+       */
+      CFBundleName?: string
+
+      /** 编译时支持的 CPU 指令 */
+      validArchitectures: ('arm64' | 'arm64e' | 'armv7' | 'armv7s' | 'x86_64')[]
```
Comment (zguolee) [inline packages/core/src/config/types/appPlus.ts:592]: 该值应该是可选？，defaultManifestConfig有一些类型报错
https://doc.dcloud.net.cn/uni-app-x/plugin/uts-plugin.html#config-json-2，默认或许是[arm64]

File: packages/core/src/config/types/appPlus.ts [hunk 3]
```diff
@@ -0,0 +1,2149 @@
...
+
+export type AndroidPermission = `<uses-permission android:name="${AndroidPermissionName}" />`
```
Comment (zguolee) [inline packages/core/src/config/types/appPlus.ts:6]:
```
export type AndroidPermission = `<uses-permission android:name="${AndroidPermissionName}" />` | `<uses-permission android:name="${AndroidPermissionName}"/>`
```

**Review by zguolee — CHANGES_REQUESTED** [review]

File: packages/core/src/config/types/appPlus.ts (hunk 3 above)
Comment (zguolee) [reply packages/core/src/config/types/appPlus.ts:6]: `manifest.json`中permissions无空格，schema.json中有空格。

File: packages/core/src/config/types/mpWeixin.ts (hunk 1 above)
Comment (ModyQyW) [reply packages/core/src/config/types/mpWeixin.ts:10]: 在 L55，看漏了？

File: packages/core/src/config/types/appPlus.ts (hunk 2 above)
Comment (ModyQyW) [reply packages/core/src/config/types/appPlus.ts:592]: 应该是可选，但不确定默认是不是 arm64，只有这里有写默认值，我不确定这是不是全局的

File: packages/core/src/config/types/appPlus.ts (hunk 3 above)
Comment (ModyQyW) [reply packages/core/src/config/types/appPlus.ts]: 已经调整了 `AndroidPermission` 类型并重新生成

File: packages/core/src/config/types/mpWeixin.ts (hunk 1 above)
Comment (zguolee) [reply packages/core/src/config/types/mpWeixin.ts:10]: Type '{ urlCheck: false; }' is missing the following properties from type '{ condition?: boolean | undefined; es6?: boolean | undefined; enhance?: boolean | undefined; es7: boolean; postcss?: boolean | undefined; minified?: boolean | undefined; minifyWXSS?: boolean | undefined; ... 29 more ...; checkSiteMap?: boolean | undefined; }': es7, ignoreDevUnusedFiles 是这个错误，看差了

Comment (ModyQyW) [reply packages/core/src/config/types/mpWeixin.ts:10]: 是我漏了 `?:` 的问题，最新版本应该修复了

**Review by zguolee — APPROVED** [review]

**Pull request merged**